instrumentation/net/http/otelhttp/test/semantic_convention_test.go

```diff
@@ -0,0 +1,142 @@
+package test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+)
+
+type lc struct {
+	t testing.TB
+}
+
+func (l *lc) Accept(log testcontainers.Log) {
+	l.t.Log(string(log.Content))
+}
+
+var docker = flag.Bool("docker", false, "Enable tests that use docker.")
+
+func TestSemanticConventions(t *testing.T) {
+	if !*docker {
+		t.Skip("Docker not enabled.")
+	}
+	wd, err := os.Getwd()
+	ctx := context.Background()
+	req := testcontainers.ContainerRequest{
+		Image:        "ghcr.io/madvikinggod/semantic-convention-checker:0.0.1",
```

Comment:
I see that https://github.com/MadVikingGod/otel-semconv-checker is a Go application.
How about simply running the HTTP server in the same process (reuse Go code) instead of running as a docker container?